src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -707,6 +708,47 @@ void ep_rt_aot_os_environment_get_utf16 (dn_vector_ptr_t *env_array)
 #endif
 }
 
+void ep_rt_aot_create_activity_id (uint8_t *activity_id, uint32_t activity_id_len)
+{
+    // We call CoCreateGuid for windows, and use a random generator for non-windows
+    STATIC_CONTRACT_NOTHROW;
+    EP_ASSERT (activity_id != NULL);
+    EP_ASSERT (activity_id_len == EP_ACTIVITY_ID_SIZE);
+#ifdef HOST_WIN32
+    CoCreateGuid (reinterpret_cast<GUID *>(activity_id));
+#else
+    if(minipal_get_cryptographically_secure_random_bytes(activity_id, activity_id_len)==-1)
```

Comment:
Any issue with this being slow? (getting secure random bytes can be)

Reply:
We have been using secure random bytes for generating Guids for reasons. This is as slow as the managed `Guid.NewGuid` and other places where we are generating Guids.

https://github.com/dotnet/runtime/blob/2ee61bbe7bb04e97159f017c4f9eaf486dc46c1f/src/libraries/System.Private.CoreLib/src/System/Guid.Unix.cs#L15-L18